src/simsopt/geo/curvehelical.py

```diff
@@ -1,69 +1,112 @@
+
+    .. math::
+        \begin{align*}
+        x &= [R_0 + r \cos(\eta)] \cos(\phi), \\
+        y &= [R_0 + r \cos(\eta)] \sin(\phi), \\
+        z &= -r \sin(\eta).
+        \end{align*}
+
+    Along the coil, the poloidal angle depends on the toroidal angle both
+    through a secular linear term and periodic Fourier terms:
 
     .. math:: 
-        \eta = m_0 \phi/l_0 + \sum_k A_k \cos(n_0 \phi k/l_0) + B_k \sin(n_0 \phi k/l_0)
+        \eta = \frac{m \phi}{l}
+        + A_0 + \sum_{k=1}^N \left[ A_k \cos(k \phi m / l) + B_k \sin(k \phi m / l) \right]
+
+    When the toroidal angle :math:`\phi` increases by :math:`2 \pi l`,
+    the poloidal angle :math:`\eta` increases by :math:`2 \pi m`, so the
+    "rotational transform" of the coil is :math:`m / l`.
+    The shape of the helical coil on the surface can be adjusted through the
+    :math:`A_k` and :math:`B_k` Fourier coefficients, which are the optimizable degrees of
+    freedom for this class.
+
+    The order in which the degrees of freedom are stored in the state vector
+    ``x`` is
+
+    .. math::
+        A_0, A_1, \ldots, A_N, B_1, \ldots, B_N
+
+    The default values of ``m``, ``ell``, ``R0``, and ``r`` correspond to the
+    coils in the Hanson-Cary and Cary-Hanson papers.
 
     Args:
-        quadpoints: number of grid points/resolution along the curve;
-        order:  number of fourier coefficients, i.e., the length of the array A (or B);
-        n0:  toroidal periodicity/number of field periods
-        l0:  number of :math:`2\pi` turns in :math:`\phi`
-        R0:  major radius
-        r0:  minor radius
-    '''
-
-    def __init__(self, quadpoints, order, n0, l0, R0, r0, **kwargs):
+        quadpoints: (int or array-like) Grid points (or number thereof) along the curve.
+        order:  Maximum (inclusive) Fourier mode number :math:`N` for :math:`A_k` and :math:`B_k`.
+        m:  Integer describing helicity of the coil.
+        ell:  Integer :math:`l` describing helicity of the coil.
+        R0:  Major radius of the toroidal surface on which the coil lies.
+        r:  Minor radius of the toroidal surface on which the coil lies.
+    """
+
+    def __init__(self, quadpoints, order, m=5, ell=2, R0=1.0, r=0.3, **kwargs):
         if isinstance(quadpoints, int):
             quadpoints = np.linspace(0, 1, quadpoints, endpoint=False)
-        def pure(dofs, points): return jaxHelicalfouriercurve_pure(
-            dofs, points, order, n0, l0, R0, r0)
+
+        def pure(dofs, points):
+            return jaxHelicalfouriercurve_pure(dofs, points, order, m, ell, R0, r)
+
         self.order = order
-        self.n0 = n0
-        self.l0 = l0
+        self.m = m
+        self.ell = ell
         self.R0 = R0
-        self.r0 = r0
-        self.coefficients = [np.zeros((order,)), np.zeros((order,))]
+        self.r = r
+        self.coefficients = np.zeros(self.num_dofs())
+        dof_names = [f"A_{i}" for i in range(order + 1)] + [f"B_{i}" for i in range(1, order + 1)]
```

Comment:
can we have a simple unit test that the dof names are what we think they are?  I think there's already some infrastructure for this (see below).

https://github.com/hiddenSymmetries/simsopt/blob/a76cec3308f192c9a9ac25cec22e7a599a7bedca/tests/geo/test_surface.py#L448